hydra-node/src/Hydra/Logging.hs

```diff
@@ -63,17 +79,15 @@ withTracer (Verbose name) action = do
           liftIO (mkEnvelop msg >>= atomically . writeTBQueue queue)
       )
 
-  mkEnvelop :: msg -> IO Value
-  mkEnvelop msg = do
+  mkEnvelop :: msg -> IO (Envelope msg)
+  mkEnvelop message = do
     timestamp <- liftIO getCurrentTime
-    threadId <- Text.drop 9 . show <$> liftIO myThreadId
-    pure $
-      object
-        [ "namespace" .= name
-        , "timestamp" .= timestamp
-        , "thread" .= threadId
-        , "message" .= msg
-        ]
+    threadId <- mkThreadId <$> liftIO myThreadId
+    pure $ Envelope{namespace, timestamp, threadId, message}
+
+  -- NOTE(AB): This is a bit contrived but we want a numeric threadId and we
+  -- get some text which we know the structure of
+  mkThreadId = fromMaybe 0 . readMaybe . Text.unpack . Text.drop 9 . show
```

Comment:
What's stopping us from having a `threadId :: Text`?

Reply:
The schema would not fit, so we would need to change both. I agree this expression is annoying hence my comment.